zips/draft-jackgavigan-defer-and-extend.rst

```diff
@@ -0,0 +1,186 @@
+capitals.
+
+"Zcash Community Advisory Panel", also called "ZCAP", refers to the panel of
+community members assembled by the Zcash Foundation and described at [#zcap]_.
+
+"Zcash Shielded Assets", also called "ZSAs", refers to the protocol changes
+described in draft ZIP 226 [#zip-0226]_ and draft ZIP 227. [#zip-0227]_ 
+
+The character § is used when referring to sections of the Zcash Protocol
+Specification. [#protocol]_
+
+The term "Mainnet" is to be interpreted as described in § 3.12 ‘Mainnet and 
+Testnet’. [#protocol-networks]_
+
+"Lockbox" refers to the in-protocol lockbox described in ZIP 1015, [#zip-1015]_ 
+and specified in ZIP 2001. [#zip-2001]_
+
+"Current Dev Fund" refers to the block subsidy allocation described in ZIP 1015, 
+[#zip-1015]_ and specified in ZIP 2001. [#zip-2001]_
+
+"Zcash Community Grants", also called "ZCG", refers to the committee described 
+in [#zip-1015]_.
+
+"Bootstrap Project", also called "BP", refers to the 501(c)(3) nonprofit 
+corporation of that name.
+
+"Zcash Foundation", also called "ZF", refers to the 501(c)(3) nonprofit 
+corporation of that name.
+
+"Electric Coin Company", also called "ECC", refers to the Zerocoin Electric 
+Coin Company, LLC.
+
+
+Abstract
+========
+This ZIP proposes expressly prohibiting any disbursement of funds from the 
+Lockbox until after Zcash Shielded Assets is active on Mainnet, and extending 
+the Current Dev Fund for one year. 
+
+
+Motivation
+==========
+
+Zcash Shielded Assets (ZSAs) is arguably the single most impactful change to the 
+Zcash protocol since the launch of Zcash in late 2016. There is clear consensus
+support for ZSAs across the Zcash community, and a ZCAP poll carried out in 
+August 2021 indicated that the Zcash community believed that ZSAs should be 
+first priority for Zcash following the activation of NU5. [#priorities]_
+
+In January 2022, ZCG provided funding for QEDIT to research and implement ZSAs 
+[#zsagrant]_, and the functionality was demonstrated using a fork of zcashd 
+during Zcon4 in August 2023. [#zsademo]
+
+Due to ECC's decision to no longer support new protocol features in the zcashd 
+node implementation, zcashd must be deprecated before ZSAs can activate on 
+Mainnet. Zcashd deprecation was expected to be complete by April 2025. However, 
+the necessary work was not prioritized. [#nu7-timeline]_ As a result, as of March 
+2025, there is no confirmed schedule for zcashd deprecation (and, therefore, no 
+schedule for the activation of ZSAs on Mainnet). 
+
+The activation of ZSAs on Mainnet is also dependent on the merging of necessary 
+code changes into the core Zcash cryptographic libraries and the new Zcash stack 
+(Zebra, Zaino and Zallet) that will replace zcashd. With limited resources, 
+anything that distracts the engineers who are working on the new stack is likely 
+to delay ZSA activation further. 
+
+In addition, any changes to the core Zcash cryptographic libraries or Zebra -- 
```

Comment:
Any kind of changes will imply that QEDIT will have to rebase their work regardless of what they are. Although, how code changes impact a fork depends a lot on where those changes are made in the codebase and how the fork changed are coded. It could be possible that a lot of changes are actually "fast-fowardable". 

Could @vivek-arte or someone else from QEDIT comment on this paragraph?

Reply:
It is simply not true that Qedit's work would have any significant code conflicts with a lockbox distribution mechanism. They are completely independent in terms of the affected code, beyond absolutely trivial issues (conflicts in changelogs, dependency lockfiles, etc) that would be easily resolved.

Reply:
> The concern is that implementing, reviewing and merging the changes required to enable Lockbox funds distribution is likely to distract core engineers from reviewing and merging QEDIT's pull requests, and any other work that enabling ZSAs is dependent on (including implemention of the new stack -- a prerequisite for zcashd deprecation, and therefore for enabling ZSAs -- and subsequent implementing of any changes to that stack to enable support for ZSAs).
> 
> To put it another way - the issue isn't code conflicts; it's resource, time, and prioritisation conflict.

This makes it clear: you're attempting to use the ZIP process to constrain the activity of independent actors in the Zcash community. This is not a valid use of the ZIP process.

Reply:
For the record, that is not correct. If the community chooses to keep the Lockbox funds locked up until ZSAs are activated on Mainnet, "independent actors" may still do whatever they want. 

If those "independent actors" decide that they are going to boycott or refuse to assist with the deprecation of zcashd, and the deployment of ZSAs in protest at the community's decision, the community can react accordingly (e.g. by deciding to issue grants to other parties who can move those efforts forward).

Reply:
I specifically requested a comment on this "rebasing cost" on the arborist call of April 17th open discussion section. And They commented that they are up-to-date with most upstream changes and wrapping up the ones on librustzcash around PCZT. Don't trust, verify and hear it yourselves there :).